examples/hello-flake/flake.nix

```
@@ -0,0 +1,27 @@
+{
+  description = "An example devShell with flakes";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, rust-overlay, flake-utils, ... }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let 
+        overlays = [ (import rust-overlay) ];
+        pkgs = import nixpkgs {
+          inherit system overlays;
+        };
+      in
+        devShells.default = mkShell {
+          buildInputs = with pkgs; [
+            rust-bin.stable.latest.default
```

Wouldn't it be better to use:

```suggestion
            rust-bin.stable.fromRustupToolchainFile ./rust-toolchain.toml
```
?

This would lead to result more consistent with non-Nix users (i.e., there would be the single source of truth about toolchain) and also solve the problem of easier configuring additional components (which are also specifyable in `rust-toolchain.toml`)